buildSrc/src/main/java/org/apache/cassandra/gradle/testextensions/RepeatingTestExecutor.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.gradle.testextensions
+
+import org.gradle.api.internal.tasks.testing.JvmTestExecutionSpec
+import org.gradle.api.internal.tasks.testing.TestExecuter
+import org.gradle.api.internal.tasks.testing.TestResultProcessor
+import org.gradle.api.tasks.testing.Test
+import java.util.*
+
+internal class RepeatingTestExecutor(private val delegate: TestExecuter<JvmTestExecutionSpec>, private val repetitions: Int) : TestExecuter<JvmTestExecutionSpec> {
+    override fun execute(testExecutionSpec: JvmTestExecutionSpec, testResultProcessor: TestResultProcessor) {
+        val forkOptions = testExecutionSpec.javaForkOptions
+        val savedSystemProps: Map<String, Any?> = HashMap(forkOptions.systemProperties)
+        try {
+            for (i in 0 until repetitions) {
+                delegate.execute(testExecutionSpec, testResultProcessor)
```

Comment:
We'd need to wrap the processor here in order to avoid sending started/completed events for the test root multiple times, something like:

```kotlin
import org.gradle.api.internal.tasks.testing.TestCompleteEvent
import org.gradle.api.internal.tasks.testing.TestDescriptorInternal
import org.gradle.api.internal.tasks.testing.TestResultProcessor
import org.gradle.api.internal.tasks.testing.TestStartEvent
import org.gradle.api.tasks.testing.TestOutputEvent

internal class RepeatingTestResultProcessor(val delegate: TestResultProcessor): TestResultProcessor by delegate {

    var lastRun: Boolean = false
    private var rootTestDescriptorId: Any? = null

    override fun started(descriptor: TestDescriptorInternal, testStartEvent: TestStartEvent) {
        if (rootTestDescriptorId == null) {
            rootTestDescriptorId = descriptor.getId()
            delegate.started(descriptor, testStartEvent)
        } else if (descriptor.getId() != rootTestDescriptorId) {
            delegate.started(descriptor, testStartEvent)
        }
    }

    override fun completed(testId: Any?, testCompleteEvent: TestCompleteEvent) {
        if (testId != rootTestDescriptorId || lastRun) {
            delegate.completed(testId, testCompleteEvent)
        }
    }
}
```

Additionally, we'd set its `lastRun` to `true` before the last `execute` repetition.